Sign in

JuanSanchezWolox opened the pull request:
## Summary
·Created login for users.
·Created schemas and middlewares for to validate data.
·Used library jsonwebtoken for created the authentication 'jwt'.

## Known Issues

## Trello Card
https://trello.com/c/ubI72RqC

JuanSanchezWolox committed:
·Created login for users.
·Created schemas and middlewares for to validate data.
·Used library jsonwebtoken for created the authentication 'jwt'.

JuanSanchezWolox committed:
Merge branch 'master' into sign-in

jxczz2 commented on app/controllers/users.js:
```diff
@@ -1,9 +1,30 @@
 };
+
+exports.signInUser = (req, res) => {
+  verifyCredentials(req.body.email, req.body.password)
```
I think you could use a mapper to receive the body of the request

jxczz2 commented on app/controllers/users.js:
```diff
@@ -1,9 +1,30 @@
+
+exports.signInUser = (req, res) => {
+  verifyCredentials(req.body.email, req.body.password)
+    .then(response => {
```
I think using .then is fine but it tends to generate a lot of extra code. Have you considered using async and await more often over .then? I think using await on top of .then makes the code look a lot better

jxczz2 commented on app/controllers/users.js:
```diff
@@ -1,9 +1,30 @@
+exports.signInUser = (req, res) => {
+  verifyCredentials(req.body.email, req.body.password)
+    .then(response => {
+      const Token = jwt.sign(
```
I think that this logic to create the logic could be in another place

jxczz2 commented on app/controllers/users.js:
```diff
@@ -1,9 +1,30 @@
+      );
+      res.status(200).send({ token: Token });
+    })
+    .catch(err => res.status(400).send(err));
```
for this case I would expect more personalized errors, making use of other status for example for this case a 401 seems to me to be the ideal

JuanSanchezWolox replied:
But the state 401 is when the user tries to access a one module without authentication, in this case he is entering values for get a token. I believe that state 400 is more appropriate

jxczz2 replied:
you're right leave it like this

jxczz2 commented on app/services/users.js:
```diff
@@ -16,3 +16,20 @@ exports.saveUser = async data => {
+
+exports.verifyCredentials = async (email, password) => {
+  try {
+    const user = await db.User.findOne({ where: { email } });
```
I think that here there are two logics together and they could be separated, don't you think it would be better to have a method to verify if the user exists and another to verify the user's password?

JuanSanchezWolox replied:
In this case I may created one function for get the user and other for verify email and password?

jxczz2 replied:
of course it would be much better this way

JuanSanchezWolox committed:
Feedback 'Pull Request'.
·Function getUser
·Mappers and serializers for sign in.
·Created file jwt in folder helpers

JuanSanchezWolox committed:
Merge branch 'master' into sign-in

jxczz2 reviewed (approved):
Good job 👍
